pyrobosim/pyrobosim/manipulation/grasping.py

```diff
@@ -0,0 +1,182 @@
+""" Grasping utilities. """
+
+from ..utils.pose import Pose
+
+import numpy as np
+import matplotlib.pyplot as plt
+from mpl_toolkits.mplot3d import Axes3D
+from mpl_toolkits.mplot3d.art3d import Poly3DCollection
+from transforms3d.quaternions import rotate_vector, qinverse
+
+
+class Grasp:
+    """
+    Representation of an object grasp.
+    """
+    def __init__(self, origin, direction, properties=None):
+        self.origin = origin
+        self.direction = direction
+        self.properties = properties
+
+    def add_origin(self, vec):
+        """ Adds an offset vector to the origin pose. """
+        return [self.origin.x + vec[0],
+                self.origin.y + vec[1],
+                self.origin.z + vec[2]]
+
+    def plot(self, ax, color):
+        """ Displays the grasp on an existing set of axes. """
+        d = self.properties.depth
+        h = self.properties.height / 2
+        w = self.properties.max_width / 2
+
+        qinv = qinverse(self.origin.q)
+        left_bottom_base = self.add_origin(rotate_vector([w, -h, -d], qinv))
+        left_top_base = self.add_origin(rotate_vector([w, h, -d], qinv))
+        left_bottom_tip = self.add_origin(rotate_vector([w, -h, 0], qinv))
+        left_top_tip = self.add_origin(rotate_vector([w, h, 0], qinv))
+        right_bottom_base = self.add_origin(rotate_vector([-w, -h, -d], qinv))
+        right_top_base = self.add_origin(rotate_vector([-w, h, -d], qinv))
+        right_bottom_tip = self.add_origin(rotate_vector([-w, -h, 0], qinv))
+        right_top_tip = self.add_origin(rotate_vector([-w, h, 0], qinv))
+
+        gripper_verts = [
+            # Right gripper
+            [right_bottom_base, right_bottom_tip,
+                right_top_tip, right_top_base],
+            # Left gripper
+            [left_bottom_base, left_bottom_tip,
+                left_top_tip, left_top_base],
+            # Gripper base
+            [right_bottom_base, left_bottom_base,
+                left_top_base, right_top_base]
+        ]
+        ax.add_collection3d(
+            Poly3DCollection(gripper_verts, color=color, alpha=0.8)
+        )
+
+    def __repr__(self):
+        display_str = f"Grasp:\n\t{self.origin}\n\tDirection: {self.direction}\n"
+        if self.properties is not None:
+            display_str += "\t{self.properties}\n"
+        return display_str
+
+
+class ParallelGraspProperties:
+    """
+    Representation of a parallel-jaw gripper.
+
+                  |            _________
+        _____     v           |              ^
+    ===|     |  height     ===|          max_width
+        -----     ^           |_________     v
+                  |
+                              |<-depth->|
```

Comment:
Cool graphic. I understand `max_width` and `depth` very well from it. But I don't understand `height`. Is that the height of the gripper from a side view? Maybe it would help to label TOP VIEW and SIDE VIEW

Reply:
You're correct, and It's labeled in the line right below this! But adding the word "VIEW" should help with visibility